src/analysis/calculus/deriv.lean

```diff
@@ -106,38 +106,54 @@ section derivative_uniqueness
 We prove that the definitions `unique_diff_within_at` and `unique_diff_on` indeed imply the
 uniqueness of the derivative. -/
 
+/-- If a function f has a derivative f' at x, a rescaled version of f around x converges to f', i.e.,
+`n (f (x + (1/n) v) - f x)` converges to `f' v`. More generally, if `c n` tends to infinity and
+`c n * d n` tends to `v`, then `c n * (f (x + d n) - f x)` tends to `f' v`. This lemma expresses
+this fact, for functions having a derivative within a set. Its specific formulation is useful for
+tangent cone related discussions. -/
+theorem has_fderiv_within_at.lim (h : has_fderiv_within_at f f' s x)
+  {c : ℕ → 𝕜} {d : ℕ → E} {v : E} (dtop : {n : ℕ | x + d n ∈ s} ∈ (at_top : filter ℕ))
+  (clim : tendsto (λ (n : ℕ), ∥c n∥) at_top at_top)
+  (cdlim : tendsto (λ (n : ℕ), c n • d n) at_top (nhds v)) :
+  tendsto (λn, c n • (f (x + d n) - f x)) at_top (nhds (f' v)) :=
+begin
+  have at_top_is_finer : at_top ≤ comap (λ (n : ℕ), x + d n) (nhds_within (x + 0) s),
```

Comment:
Since you care about proof readability, I must say I don't like this `x + 0` which you wouldn't write on paper, and get rid of right after proving `at_op_is_finer`. I would write:
```lean
  have at_top_is_finer : at_top ≤ comap (λ (n : ℕ), x + d n) (nhds_within x s),
  { conv in (nhds_within x s) { rw ← add_zero x },
    rw [← tendsto_iff_comap, nhds_within, tendsto_inf],
    split,
    { apply tendsto_add tendsto_const_nhds (tangent_cone_at.lim_zero clim cdlim) },
    { rwa tendsto_principal } },
```
This is the same number of lines.

Reply:
Yes, much better. I had never used `conv` before...

Reply:
https://github.com/leanprover-community/mathlib/blob/master/docs/extras/conv.md